features/0809-w3c-data-integrity-credential-attachment/README.md

```diff
@@ -0,0 +1,307 @@
+
+```jsonc
+{
+  "didcomm_signed_attachment": {
+    "algs_supported": ["EdDSA"],
+    "did_methods_supported": ["key"],
+    "nonce": "b19439b0-4dc9-4c28-b796-99d17034fb5c"
+  }
+}
+```
+
+- `algs_supported` - Required. List of strings indicating the Json Web Algorithms supported by the issuer for verifying the signed attachment. The list MUST contain at least one value. The values MUST be a valid algorithm identifier as defined in the [JSON Web Signature and Encryption Algorithms](https://www.iana.org/assignments/jose/jose.xhtml#web-signature-encryption-algorithms) registry.
+- `did_methods_supported` - Required. List of strings indicating which did methods are supported by the issuer for binding the credential to the holder. Values should ONLY include the method identifier of the did method. Examples values include `key` or `web`.
+- `nonce` - Required. Nonce to be used in the request to prevent replay attacks of the signed attachment.
+
+##### Binding Proof in Request
+
+The binding proof in the request points to an appended attachment containing the signed attachment.
+
+```json
+{
+  "didcomm_signed_attachment": {
+    "attachment_id": "<@id of the attachment>"
+  }
+}
+```
+
+- `attachment_id` - The id of the appended attachment included in the request message that contains the signed attachment.
+
+###### Signed Attachment Content
+
+The attachment MUST be signed by including a signature in the `jws` field of the attachment. The data MUST be a JSON document encoded in the `base64` field of the attachment. The structure of the signed attachment is described below.
+
+**JWS Payload**
+
+```json
+{
+  "nonce": "<request_nonce>",
+  "aud": "did:key:z6MkwXG2WjeQnNxSoynSGYU8V9j3QzP3JSqhdmkHc6SaVWoT"
+}
+```
+
+- `nonce` - Required. The `nonce` from the `didcomm_signed_attachment` object within `binding_method` from the credential offer
+- `aud` - Required. Intended audience of the signed attachment. MUST be the same as the issuer identifier `issuer` or `issuer.id` from the `credential` in the offer. If the `issuer` is not included in the credential in the offer, the `aud` MUST be the same as the did of the recipient did of the DIDComm message containing the request message.
+
+**Protected Header**
+
+```json
+{
+  "alg": "EdDSA",
+  "kid": "did:key:z6MkkwiqX7BvkBbi37aNx2vJkCEYSKgHd2Jcgh4AUhi4YY1u#z6MkkwiqX7BvkBbi37aNx2vJkCEYSKgHd2Jcgh4AUhi4YY1u"
+}
+```
+
+- `alg`: REQUIRED. A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. MUST NOT be none or an identifier for a symmetric algorithm (MAC). MUST match one of the `algs_supported` entries from the offer `binding_method` object.
+- `kid`: REQUIRED. JOSE Header containing the DID URL pointing to a specific key in a did document. The did method of the DID URL MUST match with one of the `did_methods_supported` from the offer `binding_method` object.
+
+A signed attachment appended to a request message might look like this:
+
+```json
+{
```

Comment:
is this a request credential message or an issue credential message?

Reply:
A credential request message, but it seems the `format` was incorrect. Fixed now!